### PR TITLE
boards: sensortag: Add documentation.

### PR DESCRIPTION
**Pull request by ghost**
Add documentation for CC2650 SensorTag board.

### CODE REVIEW

File: boards/arm/cc2650_sensortag/doc/cc2650_sensortag.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
++-----------+------------+------------------------------------+
+
+Connections and IOs
+===================
+
+The SensorTag has one GPIO controller, along with a flexible pin
+multiplexer. In practice, the pins are routed as described in
+:file:`boards/arm/cc2650_sensortag/board.h`; the most commonly used being:
+
++-------------------------------------------------------+
```
Comment (dbkinder) [inline boards/arm/cc2650_sensortag/doc/cc2650_sensortag.rst]: This first line of the table needs the `+` for the column markers for the headings to be formatted correctly

**Review by dbkinder — CHANGES_REQUESTED** [review]
so close... one tweak for the table and we're good

**Review by pfalcon — DISMISSED** [review]
@ClumsyApe , this review comment https://github.com/zephyrproject-rtos/zephyr/pull/605#pullrequestreview-46348620 would also fit this PR:

> you should prefix your commit with "boards:"

I.e. "boards: sensortag: ...". Please don't make people guess if you apply change to Zephyr Sensor subsystem.

**Comment by ClumsyApe** [comment]
* Prefixed commit message with "boards: ".
* Fixed RST table syntax.
* Added an image.

**Comment by galak** [comment]
@dbkinder look good?

**Review by dbkinder — APPROVED** [review]
ok

**Review by galak — APPROVED** [review]

**Pull request merged**